include/mgos_i2c.h

```diff
@@ -142,6 +142,33 @@ bool mgos_i2c_write_reg_w(struct mgos_i2c *conn, uint16_t addr, uint8_t reg,
 bool mgos_i2c_write_reg_n(struct mgos_i2c *conn, uint16_t addr, uint8_t reg,
                           size_t n, const uint8_t *buf);
 
+/*
+ * Helper to set/get a number of bits in a register `reg` on a device at
+ * address `addr`.
+ * - bitoffset: 0..7 is the position at which to write `value`
+ * - bitlen   : number of bits to write
+ * - value    : the value to write there
+ *
+ * Invariants:
+ * - value must fit in `bitlen` (ie value < 2^bitlen)
+ * - bitlen+bitoffset <= register size (8 for reg_b, 16 for reg_w)
+ * - bitlen cannot be 0.
+ * - *conn cannot be NULL.
+ *
```

Comment:
can you add a couple examples here and illustrate which bits get twiddled?

Reply:
Take a look at https://github.com/mongoose-os-libs/i2c/pull/6/commits/d0d56fd0ce43f246e2597dc4775db30fae608cb6 below.